Routing fixes

nottrobin opened the pull request:
- Tidy code slightly
- Fix redirection error
- Correct URL paths for URLs without slashes

QA
--

``` bash
./run
```

Then navigate to `/account` and `/account/`, with and without being already logged in. Check everything still works, and that `/account/` is now *not* a 404.

webteam-app commented:
Starting demo at: http://snapcraft.io-pr-269.run.demo.haus/

tbille commented on app.py:
```diff
@@ -155,6 +156,20 @@
+    We like our URLs without slashes
+    """
+
+    from flask import redirect, request
```
like we did for the rest of the app, use like this:
```
flask.request ...
flask.redirect ...
```

nottrobin commented:
@tbille I've fixed the tests and added a whole load more tests. Now almost every endpoint is tested - except the ones that require authentication to reach, which is much more tricky to test.

webteam-app commented:
Starting demo at: http://snapcraft.io-pr-269.run.demo.haus/
